packages/react-async/src/tests/Async.test.tsx

```diff
@@ -17,12 +17,17 @@ jest.mock('@shopify/react-effect', () => ({
   },
 }));
 
-jest.mock('@shopify/react-intersection-observer', () => ({
-  ...require.requireActual('@shopify/react-intersection-observer'),
-  IntersectionObserver() {
-    return null;
-  },
-}));
+jest.mock('@shopify/react-intersection-observer', () =>
```

Comment:
Had to rewrite these to prevent `...` transpiling to `tslib_1.assign`.  tslib refs are considered hoisting / memory leak bugs.  We could either:
- Add tslib* to Jest's blessed variable list
- Tweak tsconfig's `target` / `module` / `importHelpers` settings (probably not feasible here; it worked for sewing-kit)